modules/sops/default.nix

```diff
@@ -134,35 +151,66 @@ in {
     sshKeyPaths = mkOption {
       type = types.listOf types.path;
       default = if config.services.openssh.enable then
-                  map (e: e.path) (lib.filter (e: e.type == "rsa") config.services.openssh.hostKeys)
-                else [];
+        map (e: e.path)
+        (lib.filter (e: e.type == "rsa") config.services.openssh.hostKeys)
+      else
+        [ ];
       description = ''
         Path to ssh keys added as GPG keys during sops description.
         This option must be explicitly unset if <literal>config.sops.sshKeyPaths</literal>.
       '';
     };
   };
-  config = mkIf (cfg.secrets != {}) {
-    assertions = [{
-      assertion = (cfg.gnupgHome == null) != (cfg.sshKeyPaths == []);
-      message = "Exactly one of sops.gnupgHome and sops.sshKeyPaths must be set";
-    }] ++ optionals cfg.validateSopsFiles (
-      concatLists (mapAttrsToList (name: secret: [{
-        assertion = builtins.pathExists secret.sopsFile;
-        message = "Cannot find path '${secret.sopsFile}' set in sops.secrets.${strings.escapeNixIdentifier name}.sopsFile";
-      } {
-        assertion =
-          builtins.isPath secret.sopsFile ||
-          (builtins.isString secret.sopsFile && hasPrefix builtins.storeDir secret.sopsFile);
-        message = "'${secret.sopsFile}' is not in the Nix store. Either add it to the Nix store or set sops.validateSopsFiles to false";
-      }]) cfg.secrets)
-    );
+  config = mkMerge [
+    (mkIf cfg.useAge {
+      system.activationScripts.generate-age-key = ''
+        if [[ ! -f "${cfg.ageKeyFile}" ]]; then
```

Comment:
How do you feel about deriving this from the ssh host key? NixOS generates ed25515 keys already.
I would also add support for that to this PR if you don't mind.

Reply:
Ok. Just read your PR text. I don't think I will have time to look into this before the weekend.

Reply:
Yeah, to reiterate, I think it would be pretty useful for user keys, while it would be of limited use for host keys. If we work on the conversion for user keys though, we might as well derive host keys too :)
As for implementation details, the age tool itself seems to make a peculiar distinction between Ed25519 (SSH) and X25519 (age-native) keys. Instead of a pure conversion, they a specific SSH key type and [add some metadata to the key wrapper](https://github.com/FiloSottile/age/blob/85763d390a45b546ad2ac34a5e3e09ccd047c6df/agessh/agessh.go#L223). I'm not sure if that's relevant for our use case, I'll need to take a close look.
Otherwise, The SSH key conversion path looks to be somewhat straightforward, though in a perfect world I would prefer that upstream sops implement native age SSH key support.

Reply:
I asked about native ssh support as well when the feature was added, but they did not want to implement and also the age maintainers that reviewed the sops PR did not encourage it. Also not sure about the ssh metadata. In theory it should not be required.

Reply:
Nevertheless. I think having an activation script generating a new age key is also something, some people prefer. So let's leave it in.